wagtail/images/blocks.py

```diff
@@ -148,9 +148,9 @@ def to_python(self, value):
     def bulk_to_python(self, values):
         values = list(values)
 
-        if any(isinstance(value, int) for value in values):
-            # `values` is a list of image IDs (as we might encounter if an ImageChooserBlock has been
-            # changed to an ImageBlock with no data migration)
+        if values and all(value is None or isinstance(value, int) for value in values):
```

Comment:
The `if values` part isn't really necessary - clearly it shouldn't matter whether we treat the empty list as an empty list of IDs or an empty list of dicts - but given my previous reasoning about the ImageChooserBlock logic being the exceptional special case, and empty lists being common, I figured it was better to handle them through the default ImageBlock code path.